index.d.ts

```diff
@@ -16,20 +18,20 @@ export type Mapper<ValueType, KeyType, MappedValueType> = (
 export default function pProps<
 	KeyType extends unknown,
 	ValueType extends unknown,
-	MappedValueType = ValueType
+	MappedValueType = PromiseResult<ValueType>
 >(
-	input: Map<KeyType, PromiseLike<ValueType> | ValueType>,
+	input: Map<KeyType, ValueType>,
 	mapper?: Mapper<ValueType, KeyType, MappedValueType>,
 	options?: Options
 ): Promise<Map<KeyType, MappedValueType>>;
 export default function pProps<
-	KeyType extends string,
-	ValueType extends unknown,
-	MappedValueType = ValueType
+	InputType extends object,
```

Comment:
You'll need to replace this with a mapped object type to get the value out of the Promises on the `input` object so that `Mapper` can receive the unwrapped type.

Reply:
@sindresorhus dislikes `object` type, you'll need to replace this with an object type with an index signature to pass review.